kdc/kerberos5.c

```diff
@@ -2143,7 +2236,26 @@ _kdc_as_rep(astgs_request_t r)
 			"%s pre-authentication succeeded -- %s",
 			pat[n].name, r->cname);
 		found_pa = 1;
+
+		/*
+		 * Set r->replaced_reply_key for PAC generation (tells
+		 * Samba to include the NT hash in the PAC for NTLM)
+		 * and in case the audit code would like a boolean
+		 * operator to know if the authentiation was PKINIT
+		 * (and similar) or ENC-TS/ENC-CHAL (where the long
+		 * term key is used).
+		 */
```

Comment:
Prefer to keep the frontend AD-agnostic to the extent possible. There is already a comment in `generate_pac()`. I think you can remove this comment.

Reply:
OK.